README.md

```diff
@@ -230,176 +247,156 @@ regexes instead. That is, if you are using a lexer, you should *not* use the
-* `formatError(token)` -> return a string with an error message describing the line/col of the offending token. You might like to include a preview of the line in question.
-* `has(tokenType)` -> return true if the lexer can emit tokens with that name. Used to resolve `%`-specifiers in compiled nearley grammars.
+nearley does this for you with a built-in lexer. However, using a specialized lexer can bring large performance gains and other benefits:
 
-If Parser isn't given a lexer option, it will look for a `.lexer` attribute on its Grammar. The `@lexer` directive allows exporting a lexer object from your `.ne` grammar file. (See `json.ne` for an example.)
+- In many cases, it makes things faster by more than an order of magnitude.
+- Allows you to write cleaner, more maintainable grammars. The idea is to explcitly list all possible token types.
+- Helps to avoid ambiguity in the grammar. A lexer can confidently tell that `className` is not keyword `class` and `Name` after it.
 
+nearley supports and recommends [Moo](https://github.com/tjvr/moo), a super-fast tokenizer. Here's a basic example:
 
-### Custom tokens
-
-Nearley assumes by default that your fundamental unit of parsing, called a
-*token*, is a character. That is, you're parsing a list of characters. However,
-sometimes you want to preprocess your string to turn it into a list of *lexical
-tokens*. This means, instead of seeing "1", "2", "3", the nearley might just
-see a single list item "123". This is called *tokenizing*, and it can bring you
-decent performance gains. It also allows you to write cleaner, more
-maintainable grammars and to prevent ambiguous grammars.
-
-Tokens can be defined in two ways: literal tokens and testable tokens. A
-literal token matches exactly, while a testable token runs a function to test
-whether it is a match or not.
-
-```
+```coffeescript
 @{%
-var print_tok  = {literal: "print"};
-var number_tok = {test: function(x) {return x.constructor === Number; }}
-%}
+const moo = require("moo");
 
-main -> %print_tok %number_tok
-```
-
-Now, instead of parsing the string `"print 12"`, you would parse the array
-`["print", 12]`.
-
-You can write your own tokenizer using regular expressions, or choose from
-several existing tokenizing libraries for node.
+const lexer = moo.compile({
+  ws:     /[ \t]+/,
+  number: /[0-9]+/,
+  times:  /\*|x/
+});
+%}
 
+# Pass your lexer object using the @lexer option:
+@lexer lexer
 
-Using a parser
---------------
+# Use %token to match any token of that type instead of "token":
+multiplication -> %number %ws %times %ws %number {% ([first, , , , second]) => first * second %}
+```
 
-nearley exposes the following API:
+Have a look at [the Moo documentation](https://github.com/tjvr/moo#usage) to learn how to use it.
 
-```js
-var grammar = require("generated-code.js");
-var nearley = require("nearley");
+You can still use raw strings, but they will only match full tokens parsed by Moo:
```

Comment:
Aside: this is (or was) actually a misfeature, but, hey, this isn't the place to discuss that. (Using raw strings is really convenient, but it interacted badly with moo's support for capture groups. Fortunately, we've removed those from Moo! :-) )